### PR TITLE
Select console items using Notifications

### PR DESCRIPTION
**Pull request by AvdLee**
I've removed a bunch of unused code inside the macOS `ConsoleListView` and integrated an option to select entities using a new notification.

@kean note that I'm still using my fork, so there's other changes in this PR from the previous one. I'll migrate my project later this week!

### CODE REVIEW

File: Sources/PulseUI/Features/Console/Views/ConsoleRouterView.swift [hunk 1]
```diff
@@ -16,6 +22,18 @@ final class ConsoleRouter: ObservableObject {
...
+#if os(macOS)
+    init() {
+        NotificationCenter.default.addObserver(self, selector: #selector(didReceiveSelectItemNotification), name: .selectNetworkConsoleEntity, object: nil)
+    }
+    
+    @objc private func didReceiveSelectItemNotification(_ notification: Notification) {
+        guard let managedObjectID = notification.object as? NSManagedObjectID else { return }
+        selection = ConsoleSelectedItem.entity(managedObjectID)
+    }
+#endif
```
Comment (AvdLee) [inline Sources/PulseUI/Features/Console/Views/ConsoleRouterView.swift:35]: This is the code regarding notification driven selection

File: Sources/PulseUI/Features/Console/List/ConsoleListView.swift [hunk 2]
```diff
@@ -135,24 +135,13 @@ private struct _ConsoleListView: View {
 #if os(macOS)
 @available(iOS 15, macOS 13, visionOS 1.0, *)
 private struct _ConsoleListView: View {
-    @EnvironmentObject private var environment: ConsoleEnvironment
-    @EnvironmentObject private var listViewModel: ConsoleListViewModel
     @EnvironmentObject private var searchViewModel: ConsoleSearchViewModel
-
-    @State private var selectedObjectID: NSManagedObjectID? // Has to use for Table
-    @State private var selection: ConsoleSelectedItem?
-    @State private var shareItems: ShareItems?
```
Comment (AvdLee) [inline Sources/PulseUI/Features/Console/List/ConsoleListView.swift:144]: A bunch of these properties were not used. I guess due to refactors over time?

File: Sources/PulseUI/Features/Console/List/ConsoleListView.swift [hunk 3]
```diff
@@ -168,7 +157,7 @@ private struct _ConsoleListView: View {
             }
             Divider()
             ScrollViewReader { proxy in
-                List(selection: $selection) {
+                List(selection: $router.selection) {
```
Comment (AvdLee) [inline Sources/PulseUI/Features/Console/List/ConsoleListView.swift:160]: By linking directly to the router, we can remove local state and keep the view in sync with the router at all times.

**Comment by kean** [comment]
As suggested in https://github.com/kean/Pulse/pull/278#issuecomment-2326156413, I recommend creating separate packages for it. I would also avoid relying on these top-level componts like the router cause they might change. It should be safe to use the individual private leaf views.

**Pull request closed without merging**